cds_ils/importer/views.py

```diff
@@ -0,0 +1,169 @@
+        return jsonify(payload), error.status
+
+    @blueprint.route('/importer/check/<log_id>', methods=['GET'])
+    @need_permissions("document-importer")
+    def check(log_id):
+        log = ImporterTaskLog.query.filter_by(id=log_id).first()
+        if log:
+            entries = ImporterTaskEntry.query \
+                .filter_by(import_id=log_id) \
+                .order_by(ImporterTaskEntry.entry_index.asc()) \
+                .all()
+            obj = {
+                "id": log_id,
+                "state": log.status.value,  # enum value
+                "start_time": arrow.get(log.start_time).isoformat(),
+                "end_time": arrow.get(log.end_time).isoformat() \
+                if log.end_time else None,
+                "original_filename": log.original_filename,
+                "provider": log.provider,  # string
+                "mode": log.mode.value,  # enum value
+                "source_type": log.source_type,  # string
+            }
+            if log.entries_count:
+                obj["total_entries"] = log.entries_count
+                obj["loaded_entries"] = len(entries)
+            reports = []
+            for entry in entries:
+                if not entry.error:
+                    reports.append({
+                        "index": entry.entry_index,
+                        "success": True,
+                        "report": {
+                            "ambiguous_documents": entry.ambiguous_documents,
+                            "ambiguous_eitems": entry.ambiguous_eitems,
+                            "created_document": entry.created_document,
+                            "created_eitem": entry.created_eitem,
+                            "updated_document": entry.updated_document,
+                            "updated_eitem": entry.updated_eitem,
+                            "deleted_eitems": entry.deleted_eitems,
+                            "series": entry.series,
+                            "fuzzy_documents": entry.fuzzy_documents,
+                        }
+                    })
+                else:
+                    reports.append({
+                        "index": entry.entry_index,
+                        "success": False,
+                        "message": entry.error,
+                    })
+            obj["reports"] = reports
+            return obj
+        else:
+            raise Exception("Task not found")  # TODO
+
+    @blueprint.route('/importer', methods=['POST'])
+    @need_permissions("document-importer")
+    def importer():
+
+        @copy_current_request_context
+        def process_data(log_id, file, source_type, provider):
+            with open(app.config["IMPORTER_UPLOADS_PATH"] + "/"
+                      + file.filename) as f:
+                import_from_xml(log_id, f, source_type, provider)
+
+        if request.files:
+            file = request.files["file"]
+            provider = request.form.get("provider", None)
```

Comment:
Please prefer `webargs` instead, you have args validation and defaults out of the box.